configure.ac

```diff
@@ -59,17 +59,17 @@ AC_CHECK_DEFINED(__clang__)
 
 case "${host_os}" in
 dnl Skip this on platforms where it is just simply busted.
-openbsd*) ;;
+ openbsd*) ;;
  darwin*) LDFLAGS="$LDFLAGS -Wl,-undefined,error" ;;
-          dnl For whatever reason, the Clang sanitizers and --no-undefined for
-          dnl shared libraries are incompatible.
-       *) if test "$ac_cv_defined___clang__" = "no" || test "${LDFLAGS#*fsanitize}" = "$LDFLAGS"
-          then
-            save_LDFLAGS="$LDFLAGS"
-            LDFLAGS="$LDFLAGS -Wl,--no-undefined"
-            AC_TRY_LINK([],[],[],[LDFLAGS="$save_LDFLAGS"])
-          fi
-          ;;
+ dnl For whatever reason, the Clang sanitizers and --no-undefined for
+ dnl shared libraries are incompatible.
+ *) if test "$ac_cv_defined___clang__" = "no" -o "${LDFLAGS#*fsanitize}" = "$LDFLAGS"
```

Comment:
FWIW I used two `test`s with `||` instead of `-o` for this because, when I was working with a set of shell scripts for something else, [ShellCheck](https://github.com/koalaman/shellcheck) emitted warnings that `-o` can potentially be problematic. I have never personally encountered issues with `-o` so if this fixes an autoconf warning I'm OK with this. The other changes look good from a glance but I can verify them soon.